CHANGELOG.md

```diff
@@ -51,14 +59,18 @@
 - Improved error experience of `Session.write_pandas` method that `NotImplementError` will be raised when called.
 - Aligned error experience with reusing a closed session in non-local execution.
 
-## 1.16.0 (TBD)
+## 1.16.0 (2024-05-07)
```

Comment:
@sfc-gh-aling @sfc-gh-mabrennan those changes are made to sync with 1.16.0 release note: https://github.com/snowflakedb/snowpark-python/releases/tag/v1.16.0